### ISSUE TITLE
incorrect no-unused-vars for jsx files

### ISSUE THREAD

**Issue by szarouski** [eslint/eslint#2562]
eslint reports `Button` and `Item` as unused vars in case when their values come from arguments array instead of function arguments. Note that `Button` and `Item` are used in the jsx markup.

Example code
```js
function __Example() {
    'use strict';

    var React = arguments[0];
    var Button = arguments[1];
    var Item = arguments[2];
    var Localization = arguments[4];

    return React.createClass({
        propTypes: {
            lang: React.PropTypes.string.isRequired
        },
        render: function () {
            var lang = this.props.lang;
            var className = 'tile';

            var list = this.props.data;
            if (list.length) {
                var title = Localization.title[lang];
                var buttonText = Localization.buttonText[lang];
                var items = list.map(function (n, key) {
                    var image = n.image;
                    return (
                        <Item image={image}></Item>
                    );
                });
                
                return (
                    <div className={className}>
                        <h3 className="title">{title}</h3>
                        <div className="list">
                            {items}
                        </div>
                        <Button
                            className="button">
                            {buttonText}
                        </Button>
                    </div>
                );
            }
        }
    });
}

//somewhere in the code
__Example(React, Button, Item, Localization);
```

**Comment by gyandeeps** [comment]
Can paste the output of eslint for the above code?
What version of eslint you are using?
What kind of parser are you using?

**Comment by szarouski** [comment]
Hi @gyandeeps, eslint version is `0.21.1`. Parser is `esprima-fb`, version `15001.1.0-dev-harmony-fb`. Output: ESLint: Button is defined but never used (no-unused-vars) ESLint: Item is defined but never used (no-unused-vars)

Thank you.

**Comment by hzoo** [comment]
Maybe checkout eslint-plugin-react? https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-uses-vars.md

**Comment by szarouski** [comment]
@hzoo thank you sir, that should work. I'll check soon and report back.

**Comment by nzakas** [comment]
Working as intended. You must use eslint-plugin-react to get this functionality.
